deltacat/types/tables.py

```diff
@@ -252,29 +267,22 @@ def write_sliced_table(
         retry=retry_if_exception_type(RetryableError),
     )
 
-    manifest_entries = ManifestEntryList()
-    table_record_count = get_table_length(table)
-
-    if max_records_per_entry is None or not table_record_count:
-        # write the whole table to a single file
-        manifest_entries = retrying(
-            write_table,
-            table,
-            f"{base_path}",  # cast any non-string arg to string
-            filesystem,
-            table_writer_fn,
-            table_writer_kwargs,
-            content_type,
-            entry_params,
-            entry_type,
-        )
-    else:
-        # iteratively write table slices
-        table_slices = table_slicer_fn(table, max_records_per_entry)
-        for table_slice in table_slices:
-            slice_entries = retrying(
+    if should_write_whole_table:
+        # Write the whole table to a single file
+        if skip_manifest_write:
+            return retrying(
+                _write_table_direct,
+                table,
+                f"{base_path}",
+                filesystem,
+                table_writer_fn,
+                table_writer_kwargs,
+                content_type,
+            )
+        else:
+            manifest_entries = retrying(
```

Comment:
It looks like most of the invocations here and below only differ by (1) the function we invoke and (2) whether we specify `entry_params` and `entry_type` or not. Given this, can we reduce all the redundant if/else branches into different invocations down to:

```suggestion

entries_or_paths = retrying(writer_fn, table, f"{base_path}", filessytem, table_writer_fn, table_writer_kwargs, content_type, *additional_args)
```
Then we can consolidate our conditional logic down to just setting `writer_fn` and `*additional_args` correctly based on input parameters.